baybe/searchspace/continuous.py

```diff
@@ -186,61 +228,153 @@ def transform(
+
+            # Remove the inactive parameters from the search space
+            subspace_without_cardinality_constraint = self._drop_parameters(
+                inactive_params_sample
+            )
+
+            # Sample from the reduced space
+            try:
+                sample = subspace_without_cardinality_constraint.sample(1)
+                samples.append(sample)
+            except Exception:
+                n_fails += 1
+
+            # Avoid infinite loop
+            if n_fails >= _MAX_CARDINALITY_SAMPLING_ATTEMPTS:
+                raise RuntimeError(
+                    f"The number of failed sampling attempts has exceeded the limit "
+                    f"of {_MAX_CARDINALITY_SAMPLING_ATTEMPTS}. "
+                    f"It appears that the feasible region of the search space is very "
+                    f"small. Please review the search space constraints."
+                )
+
+        # Combine the samples and fill in inactive parameters
+        parameter_names = [p.name for p in self.parameters]
+        return pd.concat(samples).reindex(columns=parameter_names).fillna(0.0)
+
+    def _sample_inactive_parameters(self, batch_size: int = 1) -> list[set[str]]:
+        """Sample inactive parameters according to the given cardinality constraints."""
+        inactives_per_constraint = [
+            con.sample_inactive_parameters(batch_size)
+            for con in self.constraints_cardinality
+        ]
+        return [set(chain(*x)) for x in zip(*inactives_per_constraint)]
+
     def samples_full_factorial(self, n_points: int = 1) -> pd.DataFrame:
-        """Get random point samples from the full factorial of the continuous space.
+        """Deprecated!"""  # noqa: D401
+        warnings.warn(
+            f"The method '{SubspaceContinuous.samples_full_factorial.__name__}' "
+            f"has been deprecated and will be removed in a future version. "
+            f"Use '{SubspaceContinuous.sample_from_full_factorial.__name__}' instead.",
+            DeprecationWarning,
+        )
+        return self.sample_from_full_factorial(n_points)
+
+    def sample_from_full_factorial(self, batch_size: int = 1) -> pd.DataFrame:
+        """Draw parameter configurations from the full factorial of the space.
 
         Args:
-            n_points: Number of points that should be sampled.
+            batch_size: The number of parameter configurations to be sampled.
 
         Returns:
-            A data frame containing the points as rows with columns corresponding to the
-            parameter names.
+            A dataframe containing the parameter configurations as rows with columns
+            corresponding to the parameter names.
 
         Raises:
             ValueError: If there are not enough points to sample from.
         """
         full_factorial = self.full_factorial
 
-        if len(full_factorial) < n_points:
+        if len(full_factorial) < batch_size:
```

Comment:
also here and in some other places im not sure its good to change to `batch_size`, I think its good to have this word reserved for the actual reocmmendatins. Here we have abstract spaces from which we sample